realm/app-manager/src/manager.rs

```diff
@@ -40,32 +45,48 @@ pub struct Manager {
     config: Config,
     apps: HashMap<Uuid, (Application, Box<dyn ApplicationHandler + Send + Sync>)>,
     conn: JsonFramed<VsockStream, Request, Response>,
+    sealing_factory: Box<dyn KeySealingFactory + Send + Sync>,
 }
 
 impl Manager {
     pub async fn new(config: Config) -> Result<Self> {
+        if config.requires_rsi() {
+            info!("Loading rsi kernel module");
+            insmod(RSI_KO, "").await?;
+        }
+
         let vsock = VsockStream::connect(VsockAddr::new(VMADDR_CID_HOST, config.vsock_port))
             .await
             .map_err(ManagerError::VsockConnectionError)?;
         info!("Connected to warden daemon");
 
+        info!("Initializing key sealing");
+        let sealing_factory = match &config.keysealing {
+            KeySealingType::Dummy => Box::new(DummyKeySealingFactory::new(vec![0x11, 0x22, 0x33]))
+                as Box<dyn KeySealingFactory + Send + Sync>,
+            KeySealingType::HkdfSha256(ikm_source) => {
+                Box::new(HkdfSealingFactory::new(ikm_source)?)
+                    as Box<dyn KeySealingFactory + Send + Sync>
+            }
+        };
```

Comment:
Minor: Have you considered creating factory function for that (then casting shouldn't be needed)?

Reply:
I removed the casting by specifying the type for sealing_factory. Is this OK for you?